jujugui/static/gui/src/app/components/machine-view/add-machine/add-machine.js

```diff
@@ -247,13 +247,16 @@ YUI.add('machine-view-add-machine', function() {
       @return {Object} The buttons.
     */
     _generateButtons: function() {
+      var props = this.props;
       var buttons = [{
         title: 'Cancel',
-        action: this.props.close
+        action: props.close
       }, {
-        title: this.props.unit ? 'Place' : 'Create',
+        title: props.unit ? 'Place' : 'Create',
         action: this._submitForm,
-        type: 'confirm'
+        type: 'confirm',
+        disabled:!props.unit && !props.machines && props.parentId &&
```

Comment:
Should you add a comment to this line to explain what your disabling based on. Its not clear from the conditions.